### PR TITLE
EMB-29822: Revised Appendix D for IOX Pub/Sub in Add-On protocol.

### PR DESCRIPTION
**Pull request by Norman-To**

### CODE REVIEW

File: src/hardware/addon-protocols/rs232-usb.md [hunk 1]
```diff
@@ -594,180 +594,415 @@ This is an example of binary data packets for image data transferred using the M
...
-        SUB_ACK_RESULT_TOPIC_ALREADY_SUBBED = 4;
-        // Unsubscribe fails if the topic has not been subscribed to
-        SUB_ACK_RESULT_TOPIC_NOT_SUBBED = 5;
-        // Unsubscribe fails if the subscription belongs to another IOX.
-        SUB_ACK_RESULT_SUBSCRIPTION_NOT_AVAILABLE = 6;
-    }
-    Result result = 1;
-    Topic topic = 2;
-}
+## pubsub.proto
```
Comment (dharatishah) [inline src/hardware/addon-protocols/rs232-usb.md:600]: In new MD, I replaced this wrong name.

File: src/hardware/addon-protocols/rs232-usb.md [hunk 2]
```diff
@@ -594,180 +594,415 @@ This is an example of binary data packets for image data transferred using the M
...
 
-// May be extended with metadata for each topic in the future
-message TopicInfo {
-    Topic topic = 1;
-}
+## Introduciton
 
-message TopicList {
-    repeated Topic topics = 1;
-}
```
Comment (dharatishah) [inline src/hardware/addon-protocols/rs232-usb.md:668]: Spelling.

File: src/hardware/addon-protocols/rs232-usb.md [hunk 3]
```diff
@@ -594,180 +594,415 @@ This is an example of binary data packets for image data transferred using the M
...
+An example usage of the 0x8C message to subscribe to one of the topics like TOPIC_ACCEL is as described below.
 
-// Messages sent from the IOX to the GO
-message IoxToGo {
-    oneof msg {
-        PubSubToGo pub_sub = 1;
+0x8C fields values: 
+        STX=0x02, MessageId = 0x8C, Data payload: Protobuf encoding for message IoxToGo (detailed below), ETX=0x03
+
+IoxToGo message = {
```
Comment (dharatishah) [inline src/hardware/addon-protocols/rs232-usb.md:648]: added code style, indentation etc.

File: src/hardware/addon-protocols/rs232-usb.md (hunk 2 above)
Comment (Norman-To) [reply src/hardware/addon-protocols/rs232-usb.md:668]: Corrected.

**Comment by dharatishah** [comment]
@Norman-To I gave new docs.md file, could you please use that?

**Comment by Norman-To** [comment]
> @Norman-To I gave new docs.md file, could you please use that?

@dharatishah, Please send me the link to the new docs.md

**Comment by dharatishah** [comment]
> > @Norman-To I gave new docs.md file, could you please use that?
> 
> @dharatishah, Please send me the link to the new docs.md

Gave new docs.md in chat message. Please confirm if you have received it.

File: src/hardware/addon-protocols/rs232-usb.md (hunk 1 above)
Comment (Norman-To) [reply src/hardware/addon-protocols/rs232-usb.md]: It is now `#### PUB/Sub Protocol Documentation`.

File: src/hardware/addon-protocols/rs232-usb.md (hunk 3 above)
Comment (Norman-To) [reply src/hardware/addon-protocols/rs232-usb.md]: Indented.

**Review by dharatishah — APPROVED** [review]

**Review by dharatishah — APPROVED** [review]

**Pull request merged**